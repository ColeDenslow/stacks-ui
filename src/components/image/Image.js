import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { deprecated } from 'prop-types-extra';

import './Image.css';

function Image(props) {
  const { alt, className, imgStyle, message, src, style, ...otherProps } = props;

  const classNames = cn([ 'stx-image', className ]);

  const styleValue = style || imgStyle;

  return (
    <div className={classNames}>
      <div className='stx-image--content'>
        <div className='stx-image--holder'>
          <img
            alt={alt}
            className='stx-image--image'
            src={src}
            style={styleValue}
            {...otherProps}
          />
        </div>
        { message && <div className='stx-image--message'>{ message }</div> }
      </div>
    </div>
  );
}

Image.defaultProps = {
  /*
   * This property should be mandatory, since "alt" is an essential image attribute for accessibility.
   * However, making it required through PropTypes could impact existing code that is currently using Stacks UI.
   */
  alt: 'no-name-provided',
};

Image.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.string,
  imgStyle: deprecated(PropTypes.object, 'Use the style prop instead'),
  message: PropTypes.string,
  src: PropTypes.string,
  style: PropTypes.string,
};

export default Image;
