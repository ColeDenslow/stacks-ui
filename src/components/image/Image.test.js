import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Image from './Image';

describe('Image', () => {
  test('Image renders with basic props', () => {
    const output = mount(<Image
      className='the-class'
      src='https://animage.jpg'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Image renders with a message', () => {
    const output = mount(<Image
      className='the-class'
      src='https://animage.jpg'
      message='Added'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Image renders with custom/accessibility properties', () => {
    const output = mount(<Image
      className='the-class'
      src='https://animage.jpg'
      message='Added'
      aria-label='Complementary information'
      data-additional-prop='additional-prop'
    />);

    expect(toJson(output)).toMatchSnapshot();
  });
});
