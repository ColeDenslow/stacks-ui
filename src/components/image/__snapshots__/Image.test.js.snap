// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Image renders with a message 1`] = `
<Image
  alt="no-name-provided"
  className="the-class"
  message="Added"
  src="https://animage.jpg"
>
  <div
    className="stx-image the-class"
  >
    <div
      className="stx-image--content"
    >
      <div
        className="stx-image--holder"
      >
        <img
          alt="no-name-provided"
          className="stx-image--image"
          src="https://animage.jpg"
        />
      </div>
      <div
        className="stx-image--message"
      >
        Added
      </div>
    </div>
  </div>
</Image>
`;

exports[`Image Image renders with basic props 1`] = `
<Image
  alt="no-name-provided"
  className="the-class"
  src="https://animage.jpg"
>
  <div
    className="stx-image the-class"
  >
    <div
      className="stx-image--content"
    >
      <div
        className="stx-image--holder"
      >
        <img
          alt="no-name-provided"
          className="stx-image--image"
          src="https://animage.jpg"
        />
      </div>
    </div>
  </div>
</Image>
`;

exports[`Image Image renders with custom/accessibility properties 1`] = `
<Image
  alt="no-name-provided"
  aria-label="Complementary information"
  className="the-class"
  data-additional-prop="additional-prop"
  message="Added"
  src="https://animage.jpg"
>
  <div
    className="stx-image the-class"
  >
    <div
      className="stx-image--content"
    >
      <div
        className="stx-image--holder"
      >
        <img
          alt="no-name-provided"
          aria-label="Complementary information"
          className="stx-image--image"
          data-additional-prop="additional-prop"
          src="https://animage.jpg"
        />
      </div>
      <div
        className="stx-image--message"
      >
        Added
      </div>
    </div>
  </div>
</Image>
`;
